lib/lightning/collections.ex

```diff
@@ -73,13 +73,17 @@ defmodule Lightning.Collections do
   end
 
   @spec put(Collection.t(), String.t(), String.t()) ::
-          {:ok, Item.t()} | {:error, Ecto.Changeset.t()}
+          :ok | {:error, Ecto.Changeset.t()}
   def put(%{id: collection_id}, key, value) do
-    with nil <- Repo.get_by(Item, collection_id: collection_id, key: key) do
-      %Item{}
-    end
+    %Item{}
     |> Item.changeset(%{collection_id: collection_id, key: key, value: value})
-    |> Repo.insert_or_update()
+    |> Repo.insert(
+      conflict_target: [:collection_id, :key],
+      on_conflict: [set: [value: value, updated_at: NaiveDateTime.utc_now()]]
+    )
```

Comment:
`insert_or_update` isn't able to solve conflicts targeting composite PK but it's a worthy change anyways to save one BTREE (instead of having two, one extra for the `id` that woudl be rarely used) :+1: